Sources/Brave/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1569,6 +1569,21 @@ public class BrowserViewController: UIViewController {
     }
   }
 
+  func showSNSDomainInterstitialPage(originalURL: URL, visitType: VisitType?) {
+    topToolbar.leaveOverlayMode()
+    
+    guard let tab = tabManager.selectedTab, let internalUrl = URL(string: "\(InternalURL.baseUrl)/\(SNSDomainHandler.path)?url=\(originalURL.absoluteString)") else {
```

Comment:
Do the same as `encodeReaderModeURL` :) since we need to percent encode the query parameter.

Reply:
added encoding in [2d95cd2](https://github.com/brave/brave-ios/pull/6906/commits/2d95cd23e6da39da49ec78e0902f3e966a566669)